tests/lib/tools/fs-state

```diff
@@ -77,6 +116,66 @@ restore_file() {
     fi
 }
 
+start_monitor() {
+    if [ -n "$(command -v inotifywait)" ]; then
+        # In systems like arch-linux the inotifywait command fails when the output file does not exist
+        touch /tmp/fs.output
+        touch /tmp/fs.exclude
+
+        # Monitor files and directories creation
+        SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
+
+        # $PWD is excluded because the backup dir is automatically restored
+        # .snap files created in tests/lib/snaps are excluded as they are not deleted to be reused
+        inotifywait -d -m -r -e CREATE -o /tmp/fs.output --exclude "($PWD|$PROJECT_PATH/tests/lib/snaps/.*/.*.snap)" /root /var/lib/snapd /var/snap /home "$SNAP_MOUNT_DIR"
```

Comment:
If we didn't need to support 14.04 this could be done via systemd-run, which also gives us a nice way to clean it up. TBH, maybe we don't want to use this feature on 14.04 after all?

Reply:
this code has changed, we dont use inotifywait anymore, the new approach is using tree and diff